libcontainer/container_linux.go

```diff
@@ -779,3 +780,28 @@ func (c *linuxContainer) currentState() (*State, error) {
 	}
 	return state, nil
 }
+
+func (c *linuxContainer) Kill(sigStr string) error {
+	var sig uint64
+	sigN, err := strconv.ParseUint(sigStr, 10, 5)
+	if err != nil {
+		//The signal is not a number, treat it as a string (either like
+		//KILL" or like "SIGKILL")
+		syscallSig, ok := SignalMap[strings.TrimPrefix(sigStr, "SIG")]
+		if !ok {
+			return fmt.Errorf("Invalid signal: %s", sigStr)
+		}
+		sig = uint64(syscallSig)
+	} else {
+		sig = sigN
+	}
+
+	if sig == 0 {
+		return fmt.Errorf("Invalid signal: %s", sigStr)
+	}
+	state, err := c.State()
+	if err != nil {
+		return err
+	}
+	return syscall.Kill(state.InitProcessPid, syscall.Signal(sig))
```

Comment:
@avagin 
If container is not running, when we try to get the context of container, there it will return the error status of "Container ID not exists"
If container is checkpointed, directory will exists, but you would not get the state of processes inside the container.
Signal Parsing logic added to ensure user can give in signal using numbers or they can give full name of SIGNAL in strings or they can give partial names of signals.